sensor/kubernetes/clustermetrics/cluster_metrics_test.go

```diff
@@ -62,10 +63,53 @@ func (s *ClusterMetricsTestSuite) TestMultipleNodes() {
 	s.Equal(expected, metrics)
 }
 
+func (s *ClusterMetricsTestSuite) TestOfflineMode() {
+	states := []common.SensorComponentEvent{
+		common.SensorComponentEventCentralReachable,
+		common.SensorComponentEventOfflineMode,
+		common.SensorComponentEventCentralReachable,
+	}
+	metrics := s.createNewClusterMetrics(time.Millisecond)
+	s.Require().NoError(metrics.Start())
+	defer metrics.Stop(nil)
+	for _, state := range states {
+		metrics.Notify(state)
+		s.assertOfflineMode(state, metrics)
+	}
+}
+
+func (s *ClusterMetricsTestSuite) createNewClusterMetrics(interval time.Duration) *clusterMetricsImpl {
+	metricsComponent := New(s.client)
```

Comment:
Here the ticker starts with `ticker := time.NewTicker(defaultInterval)` but you change the `defaultInterval` first 3 lines down. In order to apply the new interval, one needs to call `Notify`.

This is okay, but it would be nicer to have: 
1. Changed constructor: `New(s.client, tickInterval)`
2. Additional constructor: `NewWithTicker(s.client, time.NewTicker(...))`

Reply:
I went with option (1)